lib/optimus_prime/transformers/geoip_lookup.rb

```diff
@@ -54,11 +55,22 @@ def initialize(ip_field:, maxmind_db_url:, maxmind_db_file:)
       end
 
       def write(record)
-        push get_geoip record
+        record[@geo_field_name] = {}
+        record = get_geoip record if check_valid_ip record[@ip_field]
+        push record
       end
 
       private
 
+      def check_valid_ip(ip)
```

Comment:
You rename from `check_valid_ip` to `ip_valid?` so that we know it'll return a boolean value.